docs/designs/xref.md

```diff
@@ -4,6 +4,152 @@ Besides using file path to link to another file, DocFX also allows you to give a
 - API reference documentation is usually auto generated so it's difficult to find its file path.
 - You want to reference to files in another project without need to know its file structure.
 
+## Feature Requirements/Scenarios
+- Define an internal UID in `.md` and reference to this UID within the current repository
+  ```yaml
+  inputs:
+    docs/a.md: |
+    ---
+    title: Title from yaml header a
+    uid: a
+    description: some description
+    ---
+    docs/b.md: Link to @a
+  outputs:
+    docs/b.json: |
+        {"conceptual":"<p>Link to <a href=\"a\">Title from yaml header a</a></p>\n"}
+  ```
+  > **_Notice_**: Only title will be considered as xref property for `uid` definition in `.md` files
+- Define multiple UID with same value internally with different versions and reference to this UID without versioning within the current repository
+    - If all versions for this UID are within the same product, take the one with highest versioning respecting the referencing file
+      ```yaml
+        # v1: 1.0, 2.0, 3.0
+        # v2: 4.0, 5.0
+        # v3: 3.0 
+        # should take the highest version and respect the version of the referencing file
+        inputs:
+            docs/v1/a.md: |
+            ---
+            title: Title from v1
+            uid: a
+            ---
+            docs/v2/a.md: |
+            ---
+            title: Title from v2
+            uid: a
+            ---
+            docs/v3/b.md: Link to @a
```

Comment:
>Link to @a [](start = 26, length = 10)

actually, it depends on the metadata moniker range of the file and/or the moniker range Markdig extension. 

below samples are correct, but may be a bit simplified. we may need some detailed write-up of more cases.

Reply:
and this part only applies to internal XREF

---
In reply to: [295599235](https://github.com/dotnet/docfx/pull/4687#discussion_r295599235) [](ancestors = 295599235)

Reply:
Trying to add complete requirements to this file, so yes, this part is talking about internal UID, adding h3 for it.
And just want to have a simple example to explain the case, not having all cases. We should have more test cases in our code to cover more scenarios.